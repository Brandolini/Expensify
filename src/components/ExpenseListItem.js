import { Link } from "react-router-dom";
import React from "react";
import moment from "moment";

const ExpenseListItem = ({ description, amount, id, createdAt }) => {
  return (
    <div>
      <Link to={`/edit/${id}`}>
        <h4>{description}</h4>
      </Link>
      <p>
        amount: ${amount / 100}{" "}
        <span>
          {" "}
          -{" "}
          {moment(createdAt)
            .format("MMM Do, YYYY")
            .toString()}
        </span>
      </p>
    </div>
  );
};

export default ExpenseListItem;
