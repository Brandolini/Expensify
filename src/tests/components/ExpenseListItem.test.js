// Grab imports

import ExpenseListItem from "../../components/ExpenseListItem";
import React from "react";
import expenses from "../fixtures/expenses";
import { shallow } from "enzyme";

// Render ExpenseListItem with fixture data
// Create Snapshot
test("should render ExpenseListItem with expense", () => {
  const expense = expenses[0];
  const wrapper = shallow(<ExpenseListItem {...expense} />);
  expect(wrapper).toMatchSnapshot();
});
