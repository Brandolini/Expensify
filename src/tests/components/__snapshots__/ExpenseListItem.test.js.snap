// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with expense 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h4>
      Pet Rock
    </h4>
  </Link>
  <p>
    amount: $
    30.5
     
    <span>
       
      -
       
      Dec 31st, 1969
    </span>
  </p>
</div>
`;
